modules/ModuleBuilder/Module/StudioModule.php

```diff
@@ -40,419 +40,587 @@
+                continue;
+            }
             $class = $beanList[$mod_name];
 
-            //skip if class name is not in file list, otherwise require the bean file and create new class
-            if(!isset($beanFiles[$class]) || !file_exists($beanFiles[$class])) continue;
-            
+            /**
+             * skip if class name is not in file list, otherwise require the bean file
+             * and create new class
+             */
+            if (!isset($beanFiles[$class]) || !file_exists($beanFiles[$class])) {
+                continue;
+            }
+
             //retrieve subpanels for this bean
             require_once($beanFiles[$class]);
             $bean_class = new $class();
 
             //create new subpanel definition instance and get list of tabs
-            $spd = new SubPanelDefinitions($bean_class) ;
-            if ( isset($spd->layout_defs['subpanel_setup'][strtolower($subpanel)]['module']) ){
+            $spd = new SubPanelDefinitions($bean_class);
+            if (isset($spd->layout_defs['subpanel_setup'][strtolower($subpanel)]['module'])) {
                 $spd_arr[] = $mod_name;
             }
         }
-        return  $spd_arr;
+
+        return $spd_arr;
     }
 
-    function removeFieldFromLayouts ( $fieldName )
+    /**
+     * @param string $fieldName
+     */
+    public function removeFieldFromLayouts($fieldName)
     {
-    	require_once("modules/ModuleBuilder/parsers/ParserFactory.php");
-    	$GLOBALS [ 'log' ]->info ( get_class ( $this ) . "->removeFieldFromLayouts($fieldName)" ) ;
+        require_once("modules/ModuleBuilder/parsers/ParserFactory.php");
+        $GLOBALS ['log']->info(get_class($this) . "->removeFieldFromLayouts($fieldName)");
         $sources = $this->getViewMetadataSources();
-        $sources[] = array('type'  => MB_BASICSEARCH);
-        $sources[] = array('type'  => MB_ADVANCEDSEARCH);
-        $sources[] = array('type'  => MB_POPUPSEARCH);        
-        
-        $GLOBALS [ 'log' ]->debug ( print_r( $sources,true) ) ;
-        foreach ( $sources as $name => $defs )
-        {
+        $sources[] = array('type' => MB_BASICSEARCH);
+        $sources[] = array('type' => MB_ADVANCEDSEARCH);
+        $sources[] = array('type' => MB_POPUPSEARCH);
+
+        $GLOBALS ['log']->debug(print_r($sources, true));
+        foreach ($sources as $name => $defs) {
             //If this module type doesn't support a given metadata type, we will get an exception from getParser()
             try {
-                $parser = ParserFactory::getParser( $defs [ 'type' ] , $this->module ) ;
-                if ($parser->removeField ( $fieldName ) )
-                    $parser->handleSave(false) ; // don't populate from $_REQUEST, just save as is...
-            } catch(Exception $e){}
+                $parser = ParserFactory::getParser($defs ['type'], $this->module);
+                if ($parser->removeField($fieldName)) {
+                    $parser->handleSave(false);
```

Comment:
Added method to AbstractMetaDataParser with an log message instead of an exception, as AbstractMetaDataParser class may be extended by old or legacy customisations.